heron/tools/cli/src/python/submit.py

```diff
@@ -249,22 +241,15 @@ def submit_pex(cl_args, unknown_args, tmp_dir):
   # execute main of the topology to create the topology definition
   topology_file = cl_args['topology-file-name']
   topology_class_name = cl_args['topology-class-name']
-  try:
-    execute.heron_pex(topology_file, topology_class_name, tuple(unknown_args))
-  except Exception as ex:
-    Log.error("Error when loading a topology: %s" % str(ex))
-    return False
-
-  try:
-    launch_topologies(cl_args, topology_file, tmp_dir)
-
-  except Exception as ex:
-    return False
-
-  finally:
-    shutil.rmtree(tmp_dir)
+  msg, extra_msg, retcode = execute.heron_pex(
+      topology_file, topology_class_name, tuple(unknown_args))
+  if retcode != 0:
+    return response.TopologyDefLoadResponse(
+        status=retcode, defn_file=topology_file, err_msg=msg, extra_msg=extra_msg)
+  responses = launch_topologies(cl_args, topology_file, 'python', tmp_dir)
+  shutil.rmtree(tmp_dir)
 
-  return True
+  return responses
 
 ################################################################################
 # pylint: disable=unused-argument
```

Comment:
Yes. this `run` function still needs improvement, which I will do after I figured out how to resolve the issue @billonahill asked in https://github.com/twitter/heron/pull/1571#discussion_r90506972